src/SoapCore/CustomStringWriter.cs

```diff
@@ -0,0 +1,23 @@
+using System.IO;
+using System.Text;
+
+namespace SoapCore
+{    
+    public class CustomStringWriter : StringWriter
```

Comment:
Would it be possible to create a TextWriter that writes directly to the pipewriter, while keeping track of how much data has been written?
The perf is probably fine as it is, but if we can avoid unnecessary allocations and copying of data it's a good thing 😉

Reply:
@andersjonsson Tried this approach but something keeps breaking. Either its my unit testing or the automated tests. I guess we'll have to proceed this way.